src/input/mod.rs

```diff
@@ -691,6 +691,7 @@ impl State {
                 keyboard.with_xkb_state(self, |mut state| match action {
                     LayoutSwitchTarget::Next => state.cycle_next_layout(),
                     LayoutSwitchTarget::Prev => state.cycle_prev_layout(),
+                    LayoutSwitchTarget::ByIndex(layout) => state.set_layout(Layout(layout)),
```

Comment:
Needs to check if the index is valid

Reply:
Was thinking of that myself, but not sure what should be the correct behavior on failure?

**smithay** currently resolves this by cycling through existing indexes instead of failing.

For example if a user has 3 layouts configured (`us,no,ru`), running `niri msg action switch-layout 3` would just resolve to `us` instead of failing.

Reply:
I think just ignoring a request with out-of-bounds layout would be fine.

Reply:
done